test/langtools/tools/javac/valhalla/lworld-values/universal-type-variables/UniversalTVarsCompilationTests.java

```diff
@@ -671,6 +662,17 @@ primitive class Point {}
                 }
                 """
         );
+        assertFail("compiler.err.prob.found.req",
+                """
+                import java.util.function.*;
+                class Test<__universal T> {
+                    T.ref field;
+                    void foo(T t, Consumer<? super T> action) {
+                        action.accept(field = t);
```

Comment:
well action is parameterized with a wildcard so here we are assigning a ref to a wildcard

Reply:
it could probably be better to write the test as:
```
field = t;
action.accept(field = t);
```

Reply:
Ah I see. So there's two things here:
1. assignment from `t` to `field` - that works ok (with warning)
2. passing `field` (of type `T.ref`) to `action`.

On (2), `action` is subject to capture conversion, so its type is `Consumer<#CAP>` where `T <: #CAP `. So, we have to check whether `T.ref` is assignable to `#CAP`. I believe this currently fails because `T.ref` cannot be "lifted" to `#CAP`.

I think this is ok for now, but I note that, when type variables are involved, there are always two ways to prove subtyping:

* T <: S iff upper(T) <: S
* S <: T iff S <: lower(T)

It is true that, in (2) you cannot "lift" T.ref to `'#CAP`, but it is also true that you can "unlift" #CAP down to T, at which point you would have a structural comparison between T.ref and T again (which would be ok).

Whether or not this is what we want to do is a specification question - we can revisit later.

Reply:
I see, thanks for the review